From 76fc4223c8bdcf3f82a109540b8170bdbf768593 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 9 Mar 2021 15:52:54 +0530
Subject: [PATCH 728/767] octeontx2: Re-enable FLR and MBOX interrupts

commit 748940c500679b123cda545d9fa118ecca7ad66c from
git@git.assembla.com:cavium/WindRiver.linux.git

When HWVF_RST for a VF is done by AF driver in
FLR handler then the VFs parent PF driver need to
re-enable FLR and MBOX interrupts of that VF.
This is because VF related registers in PF
register space are also cleared during HWVF_RST.

Fixes: 259c3ba4("octeontx2-af: Reset MSIX config space in FLR")
Change-Id: I3943223bbce0598c38e95136d6c2a9159608ca75
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47501
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-rm/otx2_rm.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
index 00845dafa9c2..6cd09421072e 100644
--- a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
+++ b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
@@ -453,6 +453,15 @@ static void rm_pfvf_flr_handler(struct work_struct *work)
 	enable_af_mbox_int(rm->pdev);
 	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFTRPENDX(vf->vf_id / 64),
 		   BIT_ULL(vf->intr_idx));
+
+	/* Re-enable MBOX and FLR interrupt as it gets cleared
+	 * in HWVF_RST reset
+	 */
+	rm_write64(rm, BLKADDR_RVUM, 0, RVU_PF_VFFLR_INT_ENA_W1SX(vf->vf_id),
+		   BIT_ULL(vf->intr_idx));
+	rm_write64(rm, BLKADDR_RVUM, 0,
+		   RVU_PF_VFPF_MBOX_INT_ENA_W1SX(vf->vf_id / 64),
+		   BIT_ULL(vf->intr_idx));
 }
 
 static void rm_pfvf_mbox_handler_up(struct work_struct *work)
-- 
2.31.1

